spec/system/emojis/emoji_deny_list_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+describe "Emoji deny list", type: :system, js: true do
+  let(:topic_page) { PageObjects::Pages::Topic.new }
+  let(:composer) { PageObjects::Components::Composer.new }
+  let(:emoji_picker) { PageObjects::Components::EmojiPicker.new }
+  fab!(:admin) { Fabricate(:admin) }
+
+  before { sign_in(admin) }
+
+  describe "when editing admin settings" do
+    before { SiteSetting.emoji_deny_list = "" }
+    let(:site_settings_page) { PageObjects::Pages::AdminSettings.new }
+
+    it "should allow admin to update emoji deny list" do
+      site_settings_page.visit_category("posting")
+
+      site_settings_page.select_from_emoji_list("emoji_deny_list", "fu", false)
+      site_settings_page.select_from_emoji_list("emoji_deny_list", "poop")
+
+      expect(site_settings_page.values_in_list("emoji_deny_list")).to eq(%w[fu poop])
+    end
+  end
+
+  describe "when visiting topics" do
+    before { SiteSetting.emoji_deny_list = "monkey" }
+    fab!(:topic) { Fabricate(:topic, title: "Time for :monkey: business") }
+    fab!(:post) { Fabricate(:post, topic: topic, raw: "We have no time to :monkey: around!") }
+
+    it "should remove denied emojis from page title" do
+      topic_page.visit_topic(topic)
+      expect(page.title).to eq("Time for business - Discourse")
+    end
+
+    it "should remove denied emojis from post heading" do
+      topic_page.visit_topic(topic)
+      expect(topic_page).to have_topic_title("Time for business")
+    end
+
+    it "should not show denied emoji in post body" do
+      topic_page.visit_topic(topic)
+      expect(post).not_to have_css(".emoji[name=':monkey:']")
+    end
```

Comment:
I'll recommend merging all the assertions here into a single test group since system tests are slow.